subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformValuesInjectionIntegrationTest.groovy

```diff
@@ -64,16 +64,15 @@ class ArtifactTransformValuesInjectionIntegrationTest extends AbstractDependency
                 }
             }
             
-            @AssociatedTransformAction(MakeGreenAction)
-            interface MakeGreen {
+            interface MakeGreenParameters extends TransformParameters {
                 @Input
                 String getExtension()
                 void setExtension(String value)
             }
```

Comment:
I think parameter types used exclusively by a single transform should be declared as a nested type to the transform's type. If we do this, such types can be called `Parameters` instead of these long names.

If we agree that it's a good patter, then we should communicate it in samples and integration tests.

Reply:
That sounds good. I'll try it out.